osutil/mountentry_linux.go

```diff
@@ -48,7 +88,7 @@ func ParseMountEntry(s string) (MountEntry, error) {
 	e.Type = unescape(fields[2])
 	// Parse Options if we have at least 4 fields
 	if len(fields) > 3 {
-		e.Options = strings.Split(unescape(fields[3]), ",")
+		e.Options = SplitMountOptions(unescape(fields[3]))
```

Comment:
Parsing depends on the filesystem type. And the default parser does not allow escaping of coma. So you should use that function only when it is using overlayfs.

Reply:
And given there are 7 different parsers, we should probably have a table or something so we can easily add them.

Reply:
This is such a sad state of affairs. I bet nobody in userspace does this correctly :/

Reply:
I left a more comprehensive comment below but I changed the code here a bit to call the new function only if the "overlay" type is detected and added a TODO comment for future work.